MFVNeutralino/test/TrackMover/robust_uncerts_mm_1vtx_newmctm.py

```diff
@@ -0,0 +1,712 @@
+        bincontent_err = 0.0
+        bincontent = 1.0
+      else:
+        bincontent = s_num.GetBinContent(b)/s_den.GetBinContent(b)
+        bincontent_err = math.sqrt(bincontent*(1-bincontent)/s_den.GetBinContent(b))
+        
+      s_curve.SetBinContent(b, bincontent)
+      s_curve.SetBinError(b, bincontent_err)
+   
+    return s_curve
+
+def scaledDist(dist):
+    ### OBSOLETE ###
+    s_dist = dist.Clone()
+    for b in range(0,dist.GetNbinsX()):
+        if (s_dist.GetBinContent(b) != 0):
+          if (b == 1):
+            s_dist.SetBinContent(1, 0.0)
+            s_dist.SetBinError(1, 0.0)
+          elif (b == 2):
+            s_dist.SetBinContent(2, 0.01)
+            s_dist.SetBinError(2, 0.01)
+          else:
+            s_dist.SetBinContent(b, dist.GetBinContent(b)*0.99/dist.Integral(3,100))
+            s_dist.SetBinError(b, dist.GetBinError(b)*0.99/dist.Integral(3,100))
+    return s_dist
+
+
+def scaledTOC(sig_num, sig_den, data_curve, mc_curve):
+    s_num = sig_num.Clone()
+    s_den = sig_den.Clone()
+    s_curve = s_num.Clone()
+    s_curve.Divide(s_den)
+    
+    
+    for b in range(0,s_den.GetNbinsX()):
+        if (mc_curve.GetBinContent(b) != 0 and sig_den.GetBinContent(b) != 0 ):
+            #scale = 1*(sig_curve.GetBinContent(b)*data_curve.GetBinContent(b)/mc_curve.GetBinContent(b) > 1) + (data_curve.GetBinContent(b)/mc_curve.GetBinContent(b))*( sig_curve.GetBinContent(b)*data_curve.GetBinContent(b)/mc_curve.GetBinContent(b) < 1)
+            scale = data_curve.GetBinContent(b)/mc_curve.GetBinContent(b) 
+           
+            if (sig_num.GetBinContent(b)/sig_den.GetBinContent(b) > 1.0):
+              bincontent_err = 0.0
+              bincontent = 1.0
+            elif (sig_num.GetBinContent(b)/sig_den.GetBinContent(b) < 0.0 or sig_den.GetBinContent(b) < 0.0):
+              bincontent_err = 0.0
+              bincontent = 0.0
+            else:
+              bincontent = sig_num.GetBinContent(b)/sig_den.GetBinContent(b)
+              bincontent_err = math.sqrt(bincontent*(1-bincontent)/sig_den.GetBinContent(b));
+
+            s_curve.SetBinContent(b, bincontent*scale)
+            s_curve.SetBinError(b, bincontent_err*scale)
+    return s_curve
+    
+def cutZero(original, max_n):
+    ### OBSOLETE ###
+    curve = original.Clone()
+    for b in range(0,max_n):
+        curve.SetBinContent(b, 0)
+        curve.SetBinError(b, 0)
+    return curve
+
+#############################################################################################
+
+def shiftDIST(den, sint, fr):
+    s_den = den.Clone() #ROOT.TH1D("placeholder", "", 80, 0, 80)
+    #n_den = s_den.Integral()
+    #s_den.Scale(1/s_den.Integral())
+    for b in range(1, den.GetNbinsX()):
+        #s_den.SetBinContent(b, fr*den.GetBinContent(b+sint) + (1-fr)*den.GetBinContent(b+1+sint))
+        #s_den.SetBinError(b, np.hypot(fr*den.GetBinError(b+sint), (1-fr)*den.GetBinError(b+1+sint)))
+      if (fr + sint + 1 > 0): 
```

Comment:
This isn't the only place, but it looks like there are a lot of places that cleanup would help for readability / understandability. It's easy for bugs to sneak in when we have extraneous stuff around!